src/Symfony/Component/Serializer/Mapping/AttributeMetadataInterface.php

```diff
@@ -42,6 +42,20 @@ public function addGroup($group);
     public function getGroups();
 
     /**
+     * Sets the serialization max depth for this attribute.
+     *
+     * @param int|null $maxDepth
+     */
+    public function setMaxDepth($maxDepth);
```

Comment:
By the way, I think than introducing such interfaces was a (my) mistake. Serializer Metadata are value objects. Nobody will (nor should) implement these interfaces. Composition is better for "extending" metadata.

What do you think about marking them as `@internal` (and even deprecated) @symfony/deciders?